test/e2e/node/oomkiller.go

```diff
@@ -0,0 +1,132 @@
+	podSpec       *v1.Pod
+	containerName string
+}
+
+var _ = SIGDescribe("OOMKiller [LinuxOnly]", func() {
+	f := framework.NewDefaultFramework("oomkiller-test")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	containerName := "oomkill-target-container"
+	oomPodSpec := getOOMTargetPod("oomkill-target-pod", containerName)
+	runOomKillerTest(f, oomTarget{podSpec: oomPodSpec, containerName: containerName})
+})
+
+func runOomKillerTest(f *framework.Framework, oomTarget oomTarget) {
+	ginkgo.Context("", func() {
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("setting up the pod to be used in the test")
+			e2epod.NewPodClient(f).Create(oomTarget.podSpec)
+		})
+
+		ginkgo.It("The containers terminated by OOM killer should have the reason set to OOMKilled", func() {
+
+			ginkgo.By("Waiting for the pod to be failed")
+			e2epod.WaitForPodCondition(f.ClientSet, f.Namespace.Name, oomTarget.podSpec.Name, "failed", time.Minute, func(pod *v1.Pod) (bool, error) {
+				switch pod.Status.Phase {
+				case v1.PodSucceeded:
+					return true, errors.New("pod succeeded unexpectedly")
+				case v1.PodFailed:
+					return true, nil
+				}
+				return false, nil
+			})
+
+			ginkgo.By("Fetching the latest pod status")
+			pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), oomTarget.podSpec.Name, metav1.GetOptions{})
+			if err != nil {
+				framework.Failf("Failed to get the recent pod object for name: %q", pod.Name)
+			}
+
+			ginkgo.By("Looking for the OOM target container")
+			container := getContainerByName(&pod.Status, oomTarget.containerName)
+			if container == nil {
+				framework.Failf("OOM target pod %q, container %q does not have the expected state terminated", pod.Name, container.Name)
+			}
+
+			ginkgo.By("Checking if the OOM target container is in terminated state")
+			if container.State.Terminated == nil {
+				framework.Failf("OOM target pod %q, container %q is not in the terminated state", pod.Name, container.Name)
+			}
+
+			ginkgo.By("Verifying the OOM target container has the expected reason")
+			framework.ExpectEqual(container.State.Terminated.Reason, "OOMKilled",
+				fmt.Sprintf("pod: %q, container: %q has unexpected reason: %q", pod.Name, container.Name, container.State.Terminated.Reason))
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By(fmt.Sprintf("deleting pod: %s", oomTarget.podSpec.Name))
+			e2epod.NewPodClient(f).DeleteSync(oomTarget.podSpec.Name, metav1.DeleteOptions{}, 10*time.Minute)
+		})
+	})
+}
+
+func getOOMTargetPod(podName string, ctnName string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers: []v1.Container{
+				{
+					Name:            ctnName,
+					Image:           "registry.k8s.io/stress:v1",
+					ImagePullPolicy: "Always",
+					Args:            []string{"-mem-alloc-size", "10Mi", "-mem-alloc-sleep", "1s", "-mem-total", "80Mi"},
```

Comment:
suggest to add a comment to explain reasoning behind this parameters.

Reply:
Comment added